cfme/common/__init__.py

```diff
@@ -89,3 +90,209 @@ def remove_tag(self, tag):
     def get_tags(self, tag="My Company Tags"):
         self.load_details(refresh=True)
         return mixins.get_tags(tag=tag)
+
+
+class Comparable(object):
+    """Subclassing this class enables you to have simple way to compare objects.
+
+    Compares types first, then the contents of the attributes.
+
+    .. code-block:: python
+
+       class MyClass(Comparable):
+           COMPARABLE_FIELDS = ('foo', 'bar', 'baz')
+
+    """
+    COMPARABLE_FIELDS = ()
+
+    def __eq__(self, other):
+        if not (isinstance(other, type(self)) or isinstance(self, type(other))):
+            return False
+        try:
+            for field in self.COMPARABLE_FIELDS:
+                self_field = getattr(self, field)
+                other_field = getattr(other, field)
+                if self_field != other_field:
+                    return False
+            else:
+                return True
+        except AttributeError:
+            return False
+
+
+class SummaryMixin(object):
+    """Use this mixin to have simple access to the Summary informations of an object.
+
+    Requires that the class has ``load_details(refresh)`` method defined.
+    """
+    @property
+    def summary(self):
+        if not hasattr(self, '_summary'):
+            from cfme.fixtures import pytest_selenium as sel
+            # passing sel to decouple it from the global selenium for future.
+            self._summary = Summary(sel, self)
+        return self._summary
+
+
+class Summary(object):
+    """Summary container class. An entry point to the summary listing"""
+    HEADERS = '//th[@align="left"]'
+
+    def __init__(self, sel, o):
+        self._sel = sel
+        self._object = o
+        self._keys = []
+        self.reload()
+
+    def __repr__(self):
+        return "<Summary {}>".format(" ".join(self._keys))
+
+    def reload(self):
+        self._keys = []
+        self._object.load_details(refresh=True)
+        for header in self._sel.elements(self.HEADERS):
+            header_text = self._sel.text_sane(header)
+            header_id = normalize_text(header_text, underscore=True)
```

Comment:
i think we should not reuse the normalize_text function there,
whitespace normalization and mapping text to a valid attribute name are distinct concepts
and my opinion is, that distinct concepts should have distinct functions

even if the inner implementation differs by one character ^^